src/playground-project.ts

```diff
@@ -464,7 +464,7 @@ export class PlaygroundProject extends LitElement {
     port.start();
   }
 
-  private async _getFile(name: string): Promise<SampleFile | undefined> {
+  private async _getFile(name: string): Promise<SampleFile | HttpError> {
```

Comment:
Seems weird to have an error in the completion type. Is it here because the rejection type isn't generic?

Reply:
Yeah that's the main reason.

It also feels conceptually like HTTP errors here are in the normal flow. We're essentially implementing an HTTP server, and getting e.g. a 404 because of an import of a missing project file is something that will fairly often happen due to the user's input. So, getting an exception here would be reserved for some unanticipated internal error.